### ISSUE TITLE
Optomotor Module stops with the following error

### ISSUE THREAD

**Issue by budhaChowdhury** [gilestrolab/ethoscope#122]
<img width="894" alt="Screen Shot 2020-07-14 at 3 11 24 PM" src="https://user-images.githubusercontent.com/68299423/87466710-bea63380-c5e4-11ea-8e37-3abf88e1b386.png">

The Optomotor Module starts up and runs without any problems, but after overnight tracking, exits with the error message.

**Comment by qgeissmann** [comment]
Hi @budhaChowdhury, thanks,

Does it happen systematically? Does it happen when just tracking? It looks like this is an exception when the input image (within a region of interest) is completely black (mean = 0)...
if it is transient (e.g. the camera output a few black frames to start with), we could skip the errors by changing the code to something like:

```py
try:
    scale = 128. / mean[0]
except ZeroDivisionError:
      raise NoPositionError
```

**Comment by budhaChowdhury** [comment]
Hi Quentin, 
Thanks for your feedback! 
The problem is arising when the lights in the incubator turn back on (respective day) after tracking in the dark (respective night). I just ran the module a couple more times and every single time the tracking stops as the lights are turned back on, even with a lower intensity. Do you think I should still try out the altered code?

**Comment by ggilestro** [comment]
Are you positive the IR lights are working? this is the kind of error you
would expect if IR lights were not functioning but in that case you would
expect the error to pop up shortly after lights off. If things are as you
suggest, on the other hand, I suppose what is happening is that visible
lights create a strong flash of light and the camera closes the shutter as
a response, resulting in some pitch black frames.

I don't think we've ever seen that error. We can work around it using the
solution Quentin suggested but if you have an IR lights issue you'd still
lose data.

**Comment by qgeissmann** [comment]
Yes, I would check that the lights are working as well (maybe just take a video). But I would say the *untested* patch I propose should make the tracking more robust in the case this is just a spurious couple of frames... That would perhaps be smart for us to implement so that long experiments stop after several days of acquisition... That said, it is still very hard to generate completely black frames, which is why we never had this issue...

**Comment by budhaChowdhury** [comment]
Yep the IR lights are working - I am attaching a screenshot. 
I ran a couple more times again - it quits with the same error message as the lights turn back on... Quentin can you walk me through how to use the patch? Thanks! 
ps. Should I also try a different camera? 
<img width="882" alt="Screen Shot 2020-07-14 at 5 17 36 PM" src="https://user-images.githubusercontent.com/68299423/87477500-36c92500-c5f6-11ea-95af-5bda97779d87.png">

**Comment by qgeissmann** [comment]
@ggilestro Is the update service still running on the new version? I would suggest adding the patch on a github branch so that @budhaChowdhury can update devices to this specific branch for testing...

**Comment by ggilestro** [comment]
Yes the update service runs as before but just commit this to `dev` : it's
safe enough. Thanks Quentin. Good to see you on board :)

**Comment by qgeissmann** [comment]
I don't have much infrastructure to test anything, but I am happy to help anything back-endish :). I can make a PR on dev and you can test.

**Comment by qgeissmann** [comment]
once the patch goes through to dev, you can update following [this information](https://www.notion.so/Updating-ethoscopes-0771914363c9408b988c3d62a3972b22). Ensure you pick the `dev` branch, not `master`. That would be very helpful if you let us know how it goes :),

**Comment by budhaChowdhury** [comment]
Thank you so much, will definitely keep you updated how it goes!

**Comment by budhaChowdhury** [comment]
Hi Quentin, 
in the dev branch (for ethoscope) the status says  "Software broken"- should I go ahead and update it anyway? Thanks!
ps. I am guessing I should update the Ethoscope, and not the node?

**Comment by ggilestro** [comment]
I am traveling and I have not accepted the modification request yet. Please
wait 24h.

**Comment by budhaChowdhury** [comment]
okay, just checking : ) Thanks a ton!
